ape_trezor/client.py

```diff
@@ -0,0 +1,117 @@
+from typing import Any, Dict, Optional, Tuple
+
+from eth_typing.evm import ChecksumAddress
+from trezorlib import ethereum  # type: ignore
+from trezorlib.client import get_default_client  # type: ignore
+from trezorlib.exceptions import PinException, TrezorFailure  # type: ignore
+from trezorlib.tools import parse_path as parse_hdpath  # type: ignore
+from trezorlib.transport import TransportException  # type: ignore
+
+from ape_trezor.exceptions import (
+    TrezorAccountException,
+    TrezorClientConnectionError,
+    TrezorClientError,
+)
+from ape_trezor.hdpath import HDBasePath, HDPath
+
+
+class TrezorClient:
+    """
+    This class is a client for the Trezor device.
+    """
+
+    def __init__(self, hd_root_path: HDBasePath):
+        try:
+            self._client = get_default_client()
+        except TransportException:
+            raise TrezorClientConnectionError()
+        # Handles an unhandled usb exception in Trezor transport
+        except Exception as exc:
+            raise TrezorClientError(f"Error: {exc}")
+
+        self._hd_root_path = hd_root_path
+
+    def get_account_path(self, account_id: int) -> str:
+        try:
+            return ethereum.get_address(
+                self._client, parse_hdpath(f"{self._hd_root_path}/{account_id}")
```

Comment:
```suggestion
                account_path = str(self._hd_root_path.get_account_path(account_id))
                self._client, parse_hdpath(account_path)
```

the `get_account_path()` method is better at handling when the account ID is not at the end, or at least that is the idea. Otherwise, that method `(HDPath.get_account_path())` is not getting used anywhere